[WIP] Add Dockerfiles and docker-compose configuration

dubek opened the pull request:
Running `docker compose build` and `docker compose up` will start a simple-kbs container with the backend MariaDB container.  The DB is pre-populated with a simple configuration used during CI tests of Confidential Containers with simple-kbs.

Signed-off-by: Dov Murik <dov.murik1@il.ibm.com>

----

cc: @fitzthum @dunnderr

sphrasavath reviewed (commented):
It's been a while since I've created a docker compose file, but I think you might be missing a network tag.  docker swarm needs to know if the service is connected via the host or overlay network.  I would recommend "host".  Using the default overlay network is roughly 15% slower due to NAT layers.

dubek commented:
Thanks for the suggestions @sphrasavath .

However, this small thing worked for me (without adding `networks`):

```diff
-CMD ["simple-kbs", "--grpc_sock=0.0.0.0:44444"]
+CMD ["simple-kbs", "--grpc_sock", "0.0.0.0:44444"]
```

very weird because the library we use does support both formats, and when I try locally both seem to work.

I'm looking into it to make sure I'm not imagining...

dubek commented:
I guess the network issue was PEBKAC, because even without my silly grpc_sock fix above this now works.

I removed the debug line from `Dockerfile.simple-kbs`.

As for the initial content for the DB, I added `db/initial-data.sql`. But not sure what values we should use to populate it (key? allowed measurement?). @fitzthum -- what's our take here? Maybe expose the MariaDB port to allow messing with the DB directly?

dubek commented:
1. I modified `db/initial-data.sql` to include the key needed for the @ryansavino 's CI testing.
2. I updated the PR description.

I think this is ready for review.

fitzthum reviewed (approved):
Nice. I tested this with the test image and CC installed via the operator. Works great. In the future we should add a quickstart guide or the KBS.
